source/Calamari.AzureAppService/Behaviors/TargetDiscoveryBehaviour.cs

```diff
@@ -175,6 +174,12 @@ private IEnumerable<IDeploymentSlot> ListDeploymentSlots(IWebApp webApp)
             TargetDiscoveryContext<AccountAuthenticationDetails<ServicePrincipalAccount>> context,
             TargetMatchResult matchResult)
         {
+            if (context.Authentication?.AccountId == null)
```

Comment:
I think these changes make sense, but not sure what the impetus for them was. Something about nullables?

Reply:
The `AccountId` property is marked as non-nullable and should always be provided by Server, so I'm also wondering why this check needs to be done, given we are already checking up above if the `Authentication` property is null.

Reply:
@mjhilton This change came in because Authentication and Scope became nullable on `TargetDiscoveryContext` between the Calamari.Common project AzureAppService was referencing and now. Type is used for logging the authentication type I believe, i.e. for AzureServicePrincipal authentication it's just Azure. There's a bit more context on this PR https://github.com/OctopusDeploy/Calamari/pull/901/files

Reply:
@geofflamrock The earlier null check means this should never be the case but `CreateWebAppTargetCreationServiceMessage` method expects a string for `AccountId` and with Authentication being nullable I added this for compilation. I've updated this to just check `context.Authentication` for null instead. Is there a better way to handle this?

Reply:
The methods we've added these to are private and are only called from with a code path that happens after we've check that `context.Authentication` is null, is that right? We could consider just telling the compiler that we know it isn't null at this point with `context.Authentication!.AccountId`. I'm not sure if we have any guidance on whether we should/shouldn't be doing that within the codebase, @mjhilton do you know of anything in this regard?

Reply:
It's not an area I know about or have thought about enough to have an opinion on. I don't believe we have any explicit guidance.

If you're happy with this approach, then I trust that, and am too.